### PR TITLE
feat(charts): add replica count

### PR DESCRIPTION
**Pull request by simonostendorf**
This adds configurable replicas to the helm chart.

### CODE REVIEW

**Review by apricote — CHANGES_REQUESTED** [review]
In case the user wants to deploy more than 1 replica, we also need to enable leader election so only one of the pods is actively processing at the same time.

I would suggest we add something like this to the container command:

```helm
{{- if (gt $.Values.replicaCount 1) }}
- "--leader-elect=true"
{{ -end }}
```

And document that the flag is set automatically in `values.args`  and that setting `values.replicaCount > 1` enables leader election.

**Comment by simonostendorf** [comment]
> In case the user wants to deploy more than 1 replica, we also need to enable leader election so only one of the pods is actively processing at the same time.
> 
> I would suggest we add something like this to the container command:
> 
> ```
> {{- if (gt $.Values.replicaCount 1) }}
> - "--leader-elect=true"
> {{ -end }}
> ```
> 
> And document that the flag is set automatically in `values.args` and that setting `values.replicaCount > 1` enables leader election.

Thanks for your comment @apricote 
I think the automatic leader election is a good idea. I have implemented it as requested.

File: chart/templates/deployment.yaml [hunk 1]
```diff
@@ -50,6 +50,9 @@ spec:
             - "--allocate-node-cidrs=true"
             - "--cluster-cidr={{ $.Values.networking.clusterCIDR }}"
             {{- end }}
+            {{- if (gt $.Values.replicaCount 1) }}
```
Comment (apricote) [inline chart/templates/deployment.yaml:53]: Freestyling helm templates without testing it never works, this one is on me, sorry.

```suggestion
            {{- if (gt (int $.Values.replicaCount) 1) }}
```

Current version fails e2e tests with:

```
std out err: %!(EXTRA *errors.errorString=Error: template: hcloud-cloud-controller-manager/templates/deployment.yaml:53:20: executing "hcloud-cloud-controller-manager/templates/deployment.yaml" at <gt $.Values.replicaCount 1>: error calling gt: incompatible types for comparison
```

**Review by apricote — CHANGES_REQUESTED** [review]

**Comment by apricote** [comment]
Oh and also, can you run `scripts/update-deployment-yamls.sh` and commit the changes in `deploy/`?

File: chart/templates/deployment.yaml (hunk 1 above)
Comment (simonostendorf) [reply chart/templates/deployment.yaml]: implemented, sorry i didn't saw the error

**Comment by simonostendorf** [comment]
> Oh and also, can you run `scripts/update-deployment-yamls.sh` and commit the changes in `deploy/`?

done

**Review by apricote — COMMENTED** [review]
Sorry :/

**Review by apricote — APPROVED** [review]
Thank you very much for the contribution :)

**Pull request merged**